File: examples/integrations/src/bmcModel/java/example/custommodels/ExchangeRates.java
package example.custommodels;

import org.bmc4j.Bmc;

/**
 * Proof model of the live {@link ExchangeRates} service. Same fully-qualified name as
 * the real class — so on JBMC's analysis classpath this shadows it (the real one still
 * runs in production; this is never on the runtime classpath).
 *
 * <p>Instead of a network call it returns {@link Bmc#anyInt} — a <em>symbolic</em>
 * rate. The proof then holds for <b>every</b> rate in the range at once, not just one
 * sampled value. Bound it to whatever the service can actually return.
 */
public class ExchangeRates {

    public int usdToEurBips() {
        return Bmc.anyInt(1, 20_000); // any plausible rate: 0.0001 .. 2.0
    }
}

File: examples/integrations/src/bmcModel/kotlin/example/custommodelskt/ExchangeRates.kt
package example.custommodelskt

import org.bmc4j.Bmc

/**
 * Analysis-only **model**, written in Kotlin, with the same fully-qualified name as the
 * real [example.ExchangeRates]. On JBMC's analysis classpath it shadows the real class;
 * it never reaches the test runtime classpath. The rate is symbolic, so proofs hold for
 * every value in range (0.0001 .. 2.0).
 */
class ExchangeRates {
    fun usdToEurBips(): Int = Bmc.anyInt(1, 20_000)
}
